classes/blocks/class-spreadsheet.php

```diff
@@ -42,49 +41,10 @@ public function __construct() {
 	}
 
 	/**
-	 * Get all the data that will be needed to render the block correctly.
-	 *
-	 * @param array $fields This is the array of fields of this block.
-	 *
-	 * @return array The data to be passed in the View.
+	 * Required by the `Base_Block` class.
 	 */
 	public function prepare_data( $fields ): array {
-		// Enqueue script for table filter.
-		wp_enqueue_script( 'spreadsheet-table', P4GBKS_PLUGIN_URL . 'public/js/spreadsheet.js', [ 'jquery' ], '0.2', true );
-
-		try {
-			$id = self::extract_sheet_id( $fields['url'] );
-
-			$sheet = $this->get_sheet( $id, false );
-		} catch ( \InvalidArgumentException $exception ) {
-			$sheet = null;
-		}
-
-		return [
-			'sheet'         => $sheet,
-			'css_variables' => $fields['css_variables'] ?? null,
-		];
-	}
-
-	/**
-	 * Get the ID from a google sheets publication url.
-	 *
-	 * @param string $url The url of the sheet's publication.
-	 * @return string the ID found in the sheet's publication url.
-	 * @throws \InvalidArgumentException The url is not a valid google sheets url.
-	 */
-	private static function extract_sheet_id( string $url ): string {
-		$google_sheets_pattern = '/docs\.google\.com\/spreadsheets\/d\/e\/([\w-]+)/';
-
-		$matches = [];
-		preg_match( $google_sheets_pattern, $url, $matches );
-
-		$id = $matches[1] ?? null;
-		if ( ! $id ) {
-			throw new \InvalidArgumentException( 'The url provided is not a valid public google sheets url.' );
-		}
-
-		return $id;
+		return $fields;
```

Comment:
Probably makes more sense to return an empty array here, as this function normally is't supposed to return `$fields`, even though some of our blocks do that.